lib/path.js

```diff
@@ -37,6 +37,29 @@ function normalizeArray(parts, allowAboveRoot) {
   return res;
 }
 
+// Returns an array with empty elements removed from either end of the input
+// array or the original array if no elements need to be removed
```

Comment:
I am not sure if returning the original array and a new array based on conditions is a good idea.